src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -836,7 +837,18 @@ private static bool NeedVtableOnLookupTable(SyntaxNode node)
                     node is ReturnStatementSyntax;
         }
 
-        private static EquatableArray<VtableAttribute> GetVtableAttributesToAddOnLookupTable(GeneratorSyntaxContext context)
+        private static (EquatableArray<VtableAttribute>, EquatableArray<VtableAttribute>) GetVtableAttributesToAddOnLookupTable(GeneratorSyntaxContext context)
+        {
+            // Get the lookup table assuming we are running in an authoring scenario and not in an authoring scenario

```

Comment:
> "assuming we are running in an authoring scenario and not in an authoring scenario"

I assume this is a typo? 😄

Reply:
Not really, we get the lookup table as if this was potentially an authoring component and also as if we weren't.  At this point, we don't know which one it is as we can't consult properties.  So, we return for both scenarios and then choose the right one based on properties when we can actually check it.  Updated the comment a bit to reflect that.